src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/ProjectPropertiesItemHandler.cs

```diff
@@ -10,13 +10,26 @@ namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers
     /// <summary>
     ///     Handles changes to the project and makes sure the language service is aware of them.
     /// </summary>
+    [Export(typeof(IWorkspaceContextHandler))]
     internal class ProjectPropertiesItemHandler : IEvaluationHandler
     {
-        private readonly IWorkspaceProjectContext _context;
+        private IWorkspaceProjectContext _context;
+
+        [ImportingConstructor]
+        public ProjectPropertiesItemHandler(UnconfiguredProject project)
+        {
+            Requires.NotNull(project, nameof(project));
+        }
 
-        public ProjectPropertiesItemHandler(IWorkspaceProjectContext context)
+        public string EvaluationRule
         {
-            Requires.NotNull(context, nameof(context));
+            get { return ConfigurationGeneral.SchemaName; }
+        }
+
+        public void Initialize(IWorkspaceProjectContext context)
+        {
+            if (_context != null)
+                throw new InvalidOperationException();
```

Comment:
Why is this preferable to `Requires.NotNull`?

Reply:
These are the reverse - this is checking that the field *is* still `null` when `Initialize` gets called.